pkg/healthcheck/healthcheck.go

```diff
@@ -2399,17 +2445,20 @@ func getPodStatuses(pods []corev1.Pod) map[string]map[string][]corev1.ContainerS
 	statuses := make(map[string]map[string][]corev1.ContainerStatus)
 
 	for _, pod := range pods {
-		if pod.Status.Phase == corev1.PodRunning && strings.HasPrefix(pod.Name, "linkerd-") {
+		if pod.Status.Phase == corev1.PodRunning {
```

Comment:
This existing logic here seems very confusing and overly complex and seems to have a lot of assumptions about the naming scheme of the Linkerd control plane baked in.  I think we should probably just rewrite these helper functions to be simpler.  But instead of doing that here, can we just leave this stuff alone for now and create more straightforward methods to use for the jaeger checks?  

All we need is a helper for determining if a container is running in a set of pods.  That should be as simple as iterating through the pods and their container statuses until we find a container with a matching name and then checking its status.

Reply:
makes sense!

Reply:
Why not check for the pod Status itself? 🤔 Could be simpler. I did that rn. WDYT?